python_modules/dagster/dagster/_core/event_api.py

```diff
@@ -115,9 +119,14 @@ def __new__(
         before_timestamp: Optional[float] = None,
         storage_ids: Optional[Sequence[int]] = None,
         tags: Optional[Mapping[str, Union[str, Sequence[str]]]] = None,
+        run_id: Optional[str] = None,
     ):
+        check.invariant(
+            event_type or run_id, "EventRecordsFilter must set either event_type or run_id"
+        )
```

Comment:
Can fix this with overloads I think

Reply:
On second thought, what do you think of moving this check out of the `EventRecordsFilter` object and into `get_event_records` itself? The concept of filter with neither of these set is totally coherent, so IMO it makes more sense to apply the check in the consumer rather than the data structure.

Reply:
@alangenfeld ping on question above

Reply:
i think thats fine